exchanges/bitstamp/bitstamp_test.go

```diff
@@ -871,100 +862,113 @@ func TestWsRequestReconnect(t *testing.T) {
-	if err != nil {
-		t.Fatal(err)
+	o, err := b.OHLC(context.Background(), "btcusd", start, end, "60", "10")
+	assert.NoError(t, err, "TestBitstamp_OHLC should not error")
+	assert.Equal(t, "BTC/USD", o.Data.Pair, "Pair should be correct")
+	for _, req := range o.Data.OHLCV {
+		assert.Positive(t, req.Low, "Low should be positive")
+		assert.Positive(t, req.Close, "Close should be positive")
+		assert.Positive(t, req.Open, "Open should be positive")
+		assert.Positive(t, req.Volume, "Volume should be positive")
+		assert.NotEmpty(t, req.Timestamp, "Timestamp should not be empty")
 	}
 }
 
 func TestBitstamp_GetHistoricCandles(t *testing.T) {
-	pair, err := currency.NewPairFromString("BTCUSD")
-	if err != nil {
-		t.Fatal(err)
-	}
 	start := time.Unix(1546300800, 0)
 	end := time.Unix(1577836799, 0)
-	_, err = b.GetHistoricCandles(context.Background(), pair, asset.Spot, kline.OneDay, start, end)
-	if err != nil {
-		t.Fatal(err)
+	c, err := b.GetHistoricCandles(context.Background(), btcusdPair, asset.Spot, kline.OneDay, start, end)
+	assert.NoError(t, err, "GetHistoricCandles should not error")
+	assert.Equal(t, btcusdPair, c.Pair, "Pair should be correct")
+	assert.NotEmpty(t, c, "Candles should not be empty")
+	for _, req := range c.Candles {
+		assert.Positive(t, req.High, "High should be positive")
+		assert.Positive(t, req.Low, "Low should be positive")
+		assert.Positive(t, req.Close, "Close should be positive")
+		assert.Positive(t, req.Open, "Open should be positive")
+		assert.Positive(t, req.Volume, "Volume should be positive")
+		assert.NotEmpty(t, req.Time, "Time should not be empty")
 	}
 }
 
 func TestBitstamp_GetHistoricCandlesExtended(t *testing.T) {
-	pair, err := currency.NewPairFromString("BTCUSD")
-	if err != nil {
-		t.Fatal(err)
-	}
 	start := time.Unix(1546300800, 0)
 	end := time.Unix(1577836799, 0)
 
-	_, err = b.GetHistoricCandlesExtended(context.Background(), pair, asset.Spot, kline.OneDay, start, end)
-	if err != nil {
-		t.Fatal(err)
+	c, err := b.GetHistoricCandlesExtended(context.Background(), btcusdPair, asset.Spot, kline.OneDay, start, end)
+	assert.NoError(t, err, "GetHistoricCandlesExtended should not error")
+	assert.Equal(t, btcusdPair, c.Pair, "Pair should be correct")
+	assert.NotEmpty(t, c, "Candles should not be empty")
+	for _, req := range c.Candles {
+		assert.Positive(t, req.High, "High should be positive")
+		assert.Positive(t, req.Low, "Low should be positive")
+		assert.Positive(t, req.Close, "Close should be positive")
+		assert.Positive(t, req.Open, "Open should be positive")
+		assert.Positive(t, req.Volume, "Volume should be positive")
+		assert.NotEmpty(t, req.Time, "Time should not be empty")
 	}
 }
 
 func TestGetRecentTrades(t *testing.T) {
 	t.Parallel()
+
 	currencyPair, err := currency.NewPairFromString("LTCUSD")
 	if err != nil {
-		t.Fatal(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
-	if err != nil {
-		t.Error(err)
+		t.Error("Invalid pair")
```

Comment:
assert.NoError(t, err, "NewPairFromString should not error")

Since we'll display the err to the user instead of using our own

Reply:
changed